src/tensorlake/function_executor/proto/function_executor.proto

```diff
@@ -138,6 +159,54 @@ message RunTaskResponse {
     optional SerializedObject invocation_error_output = 10;
 }
 
+// FunctionInputs contains the input data for task execution
+message FunctionInputs {
+    optional SerializedObject function_input = 1;
+    optional SerializedObject function_init_value = 2;
+}
+
+// TaskResult contains the execution outcome and outputs of a task
+message TaskResult {
+    repeated SerializedObject function_outputs = 1;
+    // Edges that the function wants the invocation to be routed to.
+    repeated string next_functions = 2;
+    optional string stdout = 3;
+    optional string stderr = 4;
+    optional bool is_reducer = 5;
+    optional Metrics metrics = 6;
+    optional TaskOutcomeCode outcome_code = 7;
+    optional TaskFailureReason failure_reason = 8;
+    // User payload for invocation error if task failed with invocation error.
+    optional SerializedObject invocation_error_output = 9;
+}
+
+// Task represents a task with its metadata and execution result
+message Task {
+    optional string task_id = 1;
+    optional string namespace = 2;
+    optional string graph_name = 3;
+    optional string graph_version = 4;
+    optional string function_name = 5;
+    optional string graph_invocation_id = 6;
+    optional string allocation_id = 7;
+    // Input data for task execution (not returned when getting/listing tasks)
```

Comment:
Should request and result be here if they are always None? If not then let's delete it to make protocol and code more clear.

Reply:
They're not always None - request is set on create, response is set when the task completes.
An alternative might be to treat the task execution as the resource being represented.

Reply:
I mean it seem like we never return a Task from FE RPCs with non None request and result. Then why are they here? It seems like something again about preferences. But I mean... It's dead code isn't it and we have to clear these fields when returning Task object. This doesn't look clean to me.

Reply:
I understand we'd need to change quite a bit of code  on FE side to address this and imo it's not worth it at this point. But I'd like to understand if I'm missing something or not here.